superset/db_engines/hive.py

```diff
@@ -48,3 +50,93 @@ def fetch_logs(self, max_rows=1024,
             if not new_logs:
                 break
         return '\n'.join(logs)
+
+def update_connect_args(url , connect_args):
+    #  set thrift_transport to handle http transport mode in hive
+    if(url.drivername == 'hive'):
+        thrift_transport = get_http_thrift_transport(url,connect_args)
+        if(thrift_transport is not None):
+            connect_args['thrift_transport'] = thrift_transport
+    else:
+        return        
+
+def get_http_thrift_transport(url , kwargs):
+    if ( 'transport_mode' in kwargs  and  kwargs['transport_mode'].lower() == 'http' ):
+        host = url.host
+        port = url.port
+        username = url.username
+        
+        password = url.password
+        if(password is None):
+           password = 'x'
+
+        #  expose kerberos specific variables and set default values HTTPKerberosAuth class 
+        http_path = 'cliservice' 
+        if('http_path' in kwargs):
+            http_path = kwargs['http_path']
+
+        #  set principal value used in http mode 
+        principal = None
+        if('principal' in kwargs):
```

Comment:
we can move this to one function so that same type of requirement, we can full-filled in one line...
        principal = None
        if('principal' in kwargs):
            principal = kwargs['principal']
like-:
function(property,**args,defaultValue)